spring-pulsar-docs/src/main/asciidoc/pulsar.adoc

```diff
@@ -1100,6 +1100,70 @@ On retry, the container sends a new batch of messages, starting with the failed
 If it fails again, it is retried, until the retries are exhausted, at which point the message is sent to the DLT.
 At that point, the message is acknowledged by the container, and the listener is handed over with the subsequent messages in the original batch.
 
+=== Pausing and Resuming Message Listener Containers
+
+There are situations in which an application might want to pause message consumption temporarily and then resume later.
+Spring for Apache Pulsar provides the ability to pause and resume the underlying message listener containers.
+When the Pulsar message listener container is paused, any polling done by the container to receive data from the Pulsar consumer will be paused.
+Similarly, when the container is resumed, the next poll starts returning data if the topic has any new records added while paused.
+The container provided pause and resume features sit a level atop of the similar capabilities provided by the Pulsar consumer which means that in addition to pausing and resuming the Pulsar consumer, the container will not even invoke the `receive` method on the consumer while it is paused.
+
+In order for applications to pause and resume the listener container, they need to inject a special bean called `PulsarListenerEndpointRegistry`.
+Once the application gets a handle on this bean, then it can query for the listener container by its id and then invoke pause and resume operations.
+
+Here is an example.
+
+====
+[source, java]
+----
+@Bean
+ApplicationRunner runner1(PulsarTemplate<String> pulsarTemplate, PulsarListenerEndpointRegistry   pulsarListenerEndpointRegistry) {
```

Comment:
TIP is a good idea, I will add it along with the extraction of the registry. I think having a complete example like the way it is there doesn't hurt and might give some extra use-case perspectives.

Reply:
The problem w/ the example is that it may quickly become stale as it is not compiled code. Also it is using `System.out` and has some other things that would not pass the code formatter etc. I think the 2nd snippet that you have added is sufficient to show how to get the container from the registry and operate on it. The pause/resume is an understandable operation that probably does not need more than that.